src/main/kotlin/org/rust/lang/core/types/infer/Fold.kt

```diff
@@ -5,10 +5,20 @@
 
 package org.rust.lang.core.types.infer
 
+import org.rust.lang.core.types.Substitution
+import org.rust.lang.core.types.re.Re
+import org.rust.lang.core.types.re.ReEarlyBound
 import org.rust.lang.core.types.ty.*
 
-typealias TypeFolder = (Ty) -> Ty
-typealias TypeVisitor = (Ty) -> Boolean
+interface TypeFolder {
+    fun foldTy(type: Ty): Ty = type
```

Comment:
Why do you use `type` name for Tys? Not only here, in multiple places. `ty` is conventional abbreviation in this project